lib/utils/types.ts

```diff
@@ -111,17 +113,20 @@ export function strictInterfaceWithOptionals<
 >(
   required: R,
   optional: O,
-  name: string
-): t.Type<t.mixed, t.InterfaceOf<R> & t.PartialOf<O>> {
+  name?: string
+): t.Type<
+  { [K in keyof R]: t.TypeOf<R[K]> } & { [K in keyof O]?: t.TypeOf<O[K]> },
```

Comment:
perhaps this deserves a little explanation :)

also you could extract this complex types and give them some names to avoid inlining them here